[FE] Feat: 공통 컴포넌트 - ListItem 구현

JaydenLee1116 opened the pull request:
#29 
- [x] ListItem 스토리북 작성
- [x] ListItem 컴포넌트 구현

JaydenLee1116 committed:
🔧 chore #21: 스토리북 스타일드 컴포넌트 설정 수정

JaydenLee1116 committed:
🔧 chore #3: 타입스크립트 path관련 디펜던시 추가

JaydenLee1116 committed:
✅ test #26: ListItem 컴포넌트 test(스토리북) 추가

JaydenLee1116 committed:
Merge branch 'feat/fe/common-component' into feat/fe/common-list-item

JaydenLee1116 committed:
🔧 chore #21: 스토리북 프리뷰 수정

- import React가 안되는 문제 해결

JaydenLee1116 committed:
✅ test #29: 스토리 수정 및 추가

JaydenLee1116 committed:
✨ feat #29: ListItem 컴포넌트 구현

hoongding commented on fe/src/components/common/ListItem/ListItem.stories.tsx:
```diff
@@ -0,0 +1,64 @@
+type Story = StoryObj<typeof meta>;
+
+export const First: Story = {
+  args: {
+    img: 'https://via.placeholder.com/150',
+    title: '파랑 선풍기',
+    town: '역삼1동',
+    timestamp: '2시간 전',
+    status: '판매중',
+    price: '24,500',
+    chatCount: 2,
+    likeCount: 5,
+    isCurrentUserItem: false,
+  },
+  // NOTE(Jayden): onClick같은 경우는 argsType을 사용해야 test시에도 사용할 수 있다.
+  argTypes: {
+    onItemClick: { action: 'onItemClick' },
+    onItemMoreClick: { action: 'onItemMoreClick' },
+  },
+};
```
Component Story Format (CSF)의 형식으로 내일 같이 공부하면서 바꿔보아요~

hoongding commented on fe/src/components/common/ListItem/index.tsx:
```diff
@@ -0,0 +1,92 @@
+  const theme = useTheme();
+
+  return (
+    <$Wrapper onClick={onItemClick}>
```
`Wrapper`라는 네이밍 보다 더 좋은 네이밍이 없을까요? 뭔가 `Wrapper`라는 네이밍만 보고선 어떤 컴포넌트의 스타일인지 한번에 느껴지지 않는것같아요..!(저번 리뷰에서 받았던 리뷰였던듯..!)
그리고 onClick말고 touch에 관한 이벤트도 내일 같이 알아보아요,,,!

hoongding replied:
제가 생각한 조금 더 좋은 네이밍으로는 `$ListItem` 이라던지 `$ListItemWrapper` 처럼 컴포넌트의 이름을 써줘도 좋을 것 같아요!

JaydenLee1116 replied:
우선 `Wrapper`가 다소 애매한 네이밍이라는 점에 동의합니다..! 고민해봤는데 `Wrapper` 혹은 `Container` 정도의 이름밖에 생각이 안나더라구요~!

또, `$ListItemWrapper`같은 형식의 경우, 어차피 `ListItem.style.ts`에서 쓰이는 `Wrapper`라면 `ListItem`이라는 표현이 중복되는 것 같아서 지웠어요! 만약 붙이게 된다면 그 하위에 있는 스타일드 컴포넌트들까지 이름에 `ListItem`을 넣어줘야할 것만 같아서요..!

훈딩은 어떻게 생각하시나욥?!

hoongding replied:
`Layout`도 좋을것 같아요~ 저는 스타일이 변수명이 길어지더라도 명확하게 쓰자는 주의라서 리뷰를 달았던 것 같아요!
서로 회의해보면서 해결해보아용!

hoongding commented on fe/src/components/common/ListItem/index.tsx:
```diff
@@ -0,0 +1,92 @@
+  onItemClick,
+  onItemMoreClick,
+}: ListItemProps) => {
+  const theme = useTheme();
```
처음 봤습니다...!ㅋㅋㅋ
저도 나중에 이렇게 써봐야겠네용ㅎㅎ

JaydenLee1116 replied:
훈딩도 보셨겠지만, Icon 컴포넌트에서 `fill`에 theme에 정리해놓은 색상을 사용하기위해 사용해보았어요!
하지만 가능하면 스타일드 컴포넌트가 아닌 리액트 컴포넌트에는 스타일에 관여하는 로직이 없었으면 했었기 때문에 더 좋은 방법은 없을지 고민이 됩니당!

훈딩이 생각하시는 좀더 좋은 방법이 있을까요?!
(당장 떠오르는건 theme을 import해와서 쓰는 것 정도..?)

hoongding commented on fe/src/components/common/ListItem/ListItem.style.ts:
```diff
@@ -0,0 +1,126 @@
+  line-height: ${({ theme }) => theme.FONT_TOKEN.FOOTNOTE.LINE_HEIGHT};
+  color: ${({ theme }) => theme.COLORS.NEUTRAL.TEXT.DEFAULT};
+
+  justify-self: end;
+  align-self: end;
```
`display`관련된 속성은 앞으로 빼면 좋겠습니다요~

hoongding reviewed (commented):
수고하셨어요! 제이든!
Styled Compoenent 재밌지요..?

JaydenLee1116 committed:
♻️ refactor #29: 코드 리뷰 반영 리팩토링

- display 관련 스타일 코드 위치 조정
- onClick: Touch관련 이벤트 변경 예정 주석 추가
- onItemMoreClick 타입 가드 처리
